runtime/gc_modron_startup/mgcalloc.cpp

```diff
@@ -516,6 +524,15 @@ J9AllocateObject(J9VMThread *vmThread, J9Class *clazz, uintptr_t allocateFlags)
 	}
 #endif /* J9VM_GC_THREAD_LOCAL_HEAP */	
 
+	if (J9_ARE_ANY_BITS_SET(vmThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_ANY)) {
+		env->saveObjects((omrobjectptr_t)objectPtr);
+		vmThread->javaVM->internalVMFunctions->internalReleaseVMAccess(vmThread);
+		vmThread->javaVM->internalVMFunctions->internalAcquireVMAccess(vmThread);
+		env->restoreObjects((omrobjectptr_t*)&objectPtr);
+	}
+
+	VM_VMAccess::clearPublicFlags(vmThread, J9_PUBLIC_FLAGS_NOT_AT_SAFE_POINT);
+
```

Comment:
In other places order is `clear flag`/`release`/`acquire`. Is order relevant?

Reply:
This clear must come after the halt check (please see new explanation in the overview).